zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -1305,7 +1363,7 @@ public void deserialize(InputArchive ia, String tag) throws IOException {
                 if (ephemeralType == EphemeralType.CONTAINER) {
                     containers.add(path);
                 } else if (ephemeralType == EphemeralType.TTL) {
-                    ttls.add(path);
+                    ttls.add(new TTLNode(path, node.stat.getCtime() + ephemeralType.getValue(node.stat.getEphemeralOwner())));
```

Comment:
How do we deal with backward compatibility?

Reply:
- `ttls` is a set only in the memory without any serializing into disk, so no backward compatibility.

> Are you an user of TTL nodes and you saw a bottle neck?
  - Look at the github description firstly
  - the ttl is a very import feature for a key-value storage system.Currently it's rusted.Look at the [ZOOKEEPER-3428](https://issues.apache.org/jira/browse/ZOOKEEPER-3428) which wants to improve the ttl further.